api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractsApiController.java

```diff
@@ -0,0 +1,57 @@
+package com.nicknathanjustin.streamercontracts.contracts;
+
+import com.nicknathanjustin.streamercontracts.contracts.requests.ContractVoteRequest;
+import com.nicknathanjustin.streamercontracts.users.UserModel;
+import com.nicknathanjustin.streamercontracts.users.UserService;
+import com.nicknathanjustin.streamercontracts.votes.VoteService;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.lang.Nullable;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/bounties")
+@RequiredArgsConstructor
+@Slf4j
+public class ContractsApiController {
+
+    @NonNull private final ContractService contractService;
+    @NonNull private final UserService userService;
+    @NonNull private final VoteService voteService;
+
+    @RequestMapping(path = "/vote", method = RequestMethod.POST)
+    public ResponseEntity voteOnContract(@RequestBody @NonNull final ContractVoteRequest contractVoteRequest,
+                                         @Nullable final OAuth2Authentication authentication) {
+        if (authentication == null) {
+            return new ResponseEntity(HttpStatus.FORBIDDEN);
+        }
+
+        final UUID contractId = contractVoteRequest.getContractId();
+        final Optional<ContractModel> optionalContractModel = contractService.getContract(contractId);
+        if (!optionalContractModel.isPresent()) {
+            log.warn("No contract found for id: {}", contractVoteRequest.getContractId());
```

Comment:
{0} instead of {}?

Reply:
I see you've used this syntax throughout the PR. Is this new syntax that I'm not aware of?

Reply:
Its based on order so no need to provide an index with {0} or {1}

I prefer parameters over concatenation because its a little better for performance -> http://blog.logscape.com/2017/03/concatenation-or-parameters-both-whats-the-top-method-of-java-logging/

Reply:
Yes agreed there, I had just never seen that syntax before. I've always seen ordinals used. But its fine by me.